effect_player.cpp

```diff
@@ -5,6 +5,26 @@
 #include <optional>
 #include <iostream>
 
+namespace bnb::oep
+{
+
+    class js_callback : public bnb::interfaces::js_callback
+    {
+    public:
+        js_callback(const oep_eval_js_result_cb& callback)
```

Comment:
why do we need a reference here? I think it is better to pass by value and move to the class member.